net/cgi-io/src/main.c

```diff
@@ -677,11 +682,11 @@ main_download(int argc, char **argv)
 		return failure(403, 0, "Requested path is not a regular file or block device");
 
 	for (p = fields[5]; p && *p; p++)
-		if (!isalnum(*p) && !strchr(" ()<>@,;:[]?.=%", *p))
+		if (!isgraph(*p) && !strchr(" ()<>@,;:[]?.=%", *p))
```

Comment:
This renders the entire check useless. Add the characters you want to whitelist to the `strchr()` expression instead. The `isgraph()` function would allow `/` for example which we explicitly want to deny here.

Reply:
Ansuel, 
You still left the isgraph there in the current version e0e109c2
(You seem to have just added the "-" to the char list, but the function was left as isgraph)

Reply:
yes sorry... forgot to change it